### DISCUSSION TITLE
Overlapping tag data allowed?

### DISCUSSION THREAD

**Discussion by nico** [saucecontrol/Compact-ICC-Profiles#15] [General]
Hi, thank you for this project and the blog post series – very interesting and informative!

I have a (mostly inconsequential) comment on https://photosauce.net/blog/post/making-a-minimal-srgb-icc-profile-part-1-trim-the-fat-abuse-the-spec . I couldn't find a comment feature over there, so I'm leaving an issue here. I hope that's fine.

The blog post says: "It’s perfectly legal for tag data to overlap, and in fact, for the TRC tags, it’s expected."

I'm not sure that's true. https://color.org/specification/ICC.1-2022-05.pdf says, in "7.3.1 Overview":

"""
Tag data elements shall not partially overlap, so there shall be no part of any tag data element that falls within
the range defined for another tag in the tag table. 

The tag table may contain multiple tags signatures that all reference the same tag data element offset, allowing
efficient reuse of tag data elements. In such cases, both the offset and size of the tag data elements in the tag
table shall be the same.
"""

So overlapping tag data is not allowed. Several tags referring to the exact same tag on the other hand is fine (that's what happens for the TRC tags).

At least that's my understanding.

**Comment by saucecontrol** [comment]
Howdy!  I'm glad you found the posts interesting.  This is a good place for that discussion anyway since it applies to the profiles here (I'll have to check on the comment feature -- it's supposed to be there).

The section you referenced is in the ICC v4.4 spec, but I can't find anything similar in the [v2 spec](https://www.color.org/ICC_Minor_Revision_for_Web.pdf), and the technique of overlapping tag data is only applied to v2 profiles since the `textDescriptionType` used in v2 has so much wasted space.

That said, I did get a report from someone who encountered issues in some Kodak commercial printing software not long after [Facebook started using my compact sRGB profile](https://www.hackerfactor.com/blog/index.php?/archives/948-Meta-Changes-at-Facebook.html).  It was rejecting the profile specifically because of the partially-overlapping tag data.  My understanding is Kodak fixed the issue because they agreed it was unusual but not strictly a violation of the spec (and because ICC's own validation tool in their [Profile Dump util](https://www.color.org/profdump.xalter) reported the profile as valid).

I suppose the v4 spec change explains why they had the check for overlapping data in place.  The specs are supposed to be backwards compatible, but clearly some restrictions were added and conventions clarified in the newer version.

Anyway, the comment is appreciated.  Since it's not an issue, I've converted this to discussion so it can stay open.  Other people may find it of interest :)

**Reply by Jaldebehot** [reply]
I also think your blog series is very well written and impressive. I applaud you. Thanks for your work! :) 

PS: It's a shame Meta is using uRGB instead of nRGB. For 8 bit precision, nRGB should be more than enough and anything more is probably overkill.
